llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp

```diff
@@ -379,6 +381,26 @@ RISCVInstructionSelector::selectSHXADD_UWOp(MachineOperand &Root,
   return std::nullopt;
 }
 
+InstructionSelector::ComplexRendererFns
+RISCVInstructionSelector::selectVLOp(MachineOperand &Root) const {
+  MachineRegisterInfo &MRI =
+      Root.getParent()->getParent()->getParent()->getRegInfo();
+  assert(Root.isReg() && "Expected operand to be a Register");
+  MachineInstr *RootDef = MRI.getVRegDef(Root.getReg());
+
+  if (RootDef->getOpcode() == TargetOpcode::G_CONSTANT &&
+      RootDef->getOperand(1).getCImm()->getSExtValue() == RISCV::VLMaxSentinel)
```

Comment:
This actually shouldn't be VLMaxSentinel. If should be `RootDef->getOperand(1).getCImm()->isAllOnesValue()`

Reply:
I considered both, like SDAG does, in the revision.

Reply:
ah, I see the comment from the other patch. Will update both.